Add a tool to check the progress of the agent migration

nicklewis opened the pull request:
The tool can be invoked with:

ruby bin/check_progress.rb /var/lib/puppet/yaml/facts

This will list the issuer of the certificate for each client whose facts
are present.

nicklewis committed:
Add a tool to check the progress of the agent migration

The tool can be invoked with:

ruby bin/check_progress.rb /var/lib/puppet/yaml/facts

This will list the issuer of the certificate for each client whose facts
are present.

jeffmccune commented:
Added a bunch of comments.   For the peanut gallery, please don't merge until we've addressed them.

-Jeff

nicklewis committed:
Support node yamldir as well as facts

When the inventory service is enabled, we don't store facts in YAML,
which means we don't have the data necessary for this script to work.
Now, it can alternately be pointed at the node yamldir, and will produce
similar results.

Also added better error handling when there are no files, as well as
when a file can't be read. In the latter case, we now simply report
'unknown' for the issuer and timestamp, rather than failing the entire
run outright.

nicklewis commented:
I pushed another commit which should address these concerns. Importantly, we now support the node dir as well as the facts dir, and will report 'unknown' for the values of nodes we are unable to read (rather than failing outright or omitting that node entirely).
